### DISCUSSION TITLE
Did HA 2024.4 just filter MQTT names removing 'room' ?

### DISCUSSION THREAD

**Discussion by arrikhan** [autoSteve/acMqtt#35] [Q&A] [answered]
I've just upgraded to HA 2024.4 and was browsing around my lighting ...

All CBUS lights that had a room assignment just lost their room prefix (ie: Bedroom 1 Light is not Light)

Anyone else seeing this odd behaviour?

![image](https://github.com/autoSteve/acMqtt/assets/13254846/a38b09ba-77c4-4b60-bd87-d5d750a3636f)

**Comment by autoSteve** [comment]
What is `removeSaFromStartOfPn = true` in _MQTT send receive_ set to? Try setting it to false, which should probably be the default.

**Comment by autoSteve** [comment]
I've altered the defaults for environment variables, because panasonic and environment sensors were true by default, now false, as well as removeSaFromStartOfPn to false.

**Comment by arrikhan** [comment]
Doesn't exist in the version of send receive i am running (previous version we worked on)

Should I upgrade and leverage function?

Interestingly it only affects devices that I have defined sa=blahblah on.

**Reply by autoSteve** [reply]
Yeah, this was always the behaviour to remove the sa= from the start of the name in earlier versions, so maybe something prior to 2024.4 was masking the behaviour. The variable is a recent addition to provide choice.

Be careful upgrading to make sure you set `entityIdAsIndentifier` to false in the environment variables. This is also a very recent change that will break your dashboards without work. This variable sets up entity naming using the identity name, and not the C-Bus network address.

**Reply by autoSteve** [reply]
Did that fix things up, @arrikhan?

**Comment by arrikhan** [comment, marked as answer]
The below sorted it first go. 

```
entityIdAsIndentifier = false
removeSaFromStartOfPn = false
```

Is it worth putting instructions in the script about removing event based script when moving to these versions? .. or did I miss that somewhere?

**Reply by autoSteve** [reply]
You don't need to remove it. The resident should disable it on start if it's named correctly.

But good call on the doco aspect.

**Reply by autoSteve** [reply]
Done. [Event script deprecation note and efficiency](https://github.com/autoSteve/acMqtt/commit/5ad1972b9d29899178d05b2e7f30141d3e5cb217)
